doc: peripheral_uart: add info about async_adapter (#7749)
nrfconnect/sdk-nrf: samples/bluetooth/peripheral_uart/README.rst

=====================

For the boards with the USB device peripheral, you can build the sample with support for the USB CDC ACM class serial port instead of the physical UART.

> This build uses the custom UART async adapter module that acts as a bridge between the USB CDC ACM and the default sample configuration.

Reviewer: between USB CDC ACM and Zephyr's UART asynchronous API used by the sample ?

See :ref:`peripheral_uart_sample_activating_variants` for details about how to build the sample with this extension using the :file:`prj_cdc.conf`.

Async adapter experimental module
   The default sample configuration uses the UART async API.

>    The UART async adapter creates and initializes an instance of the async module, so that the USB CDC ACM can provide only the interrupt interface.

Reviewer: The UART async adapter creates and initializes an instance of the async module.
This is needed, because the USB CDC ACM implementation provides only the interrupt interface.

=====================

For the boards with the USB device peripheral, you can build the sample with support for the USB CDC ACM class serial port instead of the physical UART.
This build uses the sample-specific UART async adapter module that acts as a bridge between USB CDC ACM and Zephyr's UART asynchronous API used by the sample.
See :ref:`peripheral_uart_sample_activating_variants` for details about how to build the sample with this extension using the :file:`prj_cdc.conf`.

Async adapter experimental module
   The default sample configuration uses the UART async API.
   The UART async adapter creates and initializes an instance of the async module.
   This is needed because the USB CDC ACM implementation provides only the interrupt interface.

>    The adapter uses data provided in the :c:struct:`uart_async_adapter_data` to connect to the UART device that is not using the asynchronous interface.

Reviewer 2: Just a question. Wouldn't be better just:
```suggestion
   The adapter uses data provided in the :c:struct:`uart_async_adapter_data` to connect to the UART device that does not use the asynchronous interface.
```

Author: It would as it is shorter.

   The module requires the :kconfig:option:`CONFIG_BT_NUS_UART_ASYNC_ADAPTER` to be set to ``y``.

>    For more information about the adapter, see the :file:`uart_async_adapter` source files available in the :file:`peripheral_uart/src` directory.

Reviewer 2: ```suggestion
   For more information about the adapter, see the :file:`uart_async_adapter.c` source files available in the :file:`peripheral_uart/src` directory.
```

* :ref:`nus_service_readme`
* :ref:`dk_buttons_and_leds_readme`

> * Custom :file:`uart_async_adapter` at :file:`peripheral_uart/src`

Reviewer: Is it |NCS| dependency? The code is specific to the sample
